packages/stellar_client/lib/src/client.dart

```diff
@@ -518,4 +562,363 @@ class Client {
+    }
+
+    final offers = (await _sdk.offers.forAccount(accountId).execute()).records;
+    final OfferResponse? targetOffer = offers.firstWhere(
+      (offer) => offer.id == offerId,
+      orElse: () => throw Exception(
+          'Offer with ID $offerId not found in user\'s account.'),
+    );
+
+    final Asset sellingAsset = _getAsset(sellingAssetCode);
+    final Asset buyingAsset = _getAsset(buyingAssetCode);
+
+    ManageBuyOfferOperation updateOfferOperation =
+        ManageBuyOfferOperationBuilder(
+      sellingAsset,
+      buyingAsset,
+      amount,
+      price,
+    ).setOfferId(offerId).build();
+
+    final account = await _sdk.accounts.account(accountId);
+    final Transaction transaction =
+        TransactionBuilder(account).addOperation(updateOfferOperation).build();
+    transaction.sign(_keyPair, _stellarNetwork);
+    try {
+      final SubmitTransactionResponse response =
+          await _sdk.submitTransaction(transaction);
+      if (!response.success) {
+        logger.e('Transaction failed with result: ${response.resultXdr}');
+        return false;
+      }
+      return true;
+    } catch (error) {
+      throw Exception('Transaction failed due to: ${error.toString()}');
+    }
+  }
+
+  /// Retrieves the order book for a given asset pair on the Stellar network.
+  ///
+  /// This function returns a stream of `OrderBookResponse`, which provides
+  /// real-time updates on buy and sell orders for the specified asset pair.
+  ///
+  /// ### Understanding Stellar Order Representation:
+  /// - **Price (`OrderBookResponse.asks[].price` & `OrderBookResponse.bids[].price`)**:
+  ///   Stellar stores price as `buying / selling`, meaning the displayed price
+  ///   is the **inverse** of the price provided when creating an order.
+  /// - **Amount (`OrderBookResponse.asks[].amount`)**:
+  ///   This represents the total amount of the **selling asset** available in the order book.
+  ///
+  /// ### Conversion Formula:
+  /// ```
+  /// Total selling amount = Buying amount * Price
+  /// Stored price = 1 / Provided price
+  /// ```
+  ///
+  /// ### Example:
+  /// #### **Creating an Order**
+  /// ```dart
+  /// await stellarClient.createOrder(
+  ///     sellingAssetCode: 'XLM',
+  ///     buyingAssetCode: 'TFT',
+  ///     amount: '2',     // Buying 2 TFT
+  ///     price: '0.1');   // 1 XLM = 0.1 TFT
+  /// ```
+  ///
+  /// #### **Retrieved Order Book Entry**
+  /// ```dart
+  /// OrderBookResponse {
+  ///   asks: [
+  ///     {
+  ///       amount: "0.2",   // Total selling amount = 2 * 0.1 = 0.2 XLM
+  ///       price: "10.0"    // Inverted: 1 / 0.1 = 10 XLM per TFT
+  ///     }
+  ///   ]
+  /// }
+  /// ```
+  ///
+  /// **Key Takeaways:**
+  /// - `OrderBookResponse.asks[].amount` = **Total amount of the selling asset**.
+  /// - `OrderBookResponse.asks[].price` = **Inverse of the provided price**.
+  Future<Stream<OrderBookResponse>> getOrderBook(
```

Comment:
The orderbook doesn't require being on the client. Can you make it as a helper to use it without a secret?